pkg/skaffold/deploy/kustomize.go

```diff
@@ -99,9 +100,13 @@ type KustomizeDeployer struct {
 	labels             map[string]string
 	BuildArgs          []string
 	globalConfig       string
+	useKubectl         bool
 }
 
 func NewKustomizeDeployer(runCtx *runcontext.RunContext, labels map[string]string) *KustomizeDeployer {
+	// if user has kustomize binary, prioritize that over kubectl kustomize
+	useKubectl := !kustomizeBinaryCheck() && kubectlVersionCheck(runCtx)
+
 	return &KustomizeDeployer{
 		KustomizeDeploy: runCtx.Cfg.Deploy.KustomizeDeploy,
 		kubectl: deploy.CLI{
```

Comment:
It seems odd to pass in a `kubectl.CLI` if we're not to use `kubectl`.

Reply:
I think I should rename `useKubectl` to `useKubectlKustomize`. We still need the `kubectl.CLI` as the kustomize deployer still uses `kubectl apply` either way